src/decoder.rs

```diff
@@ -197,6 +199,11 @@ impl<R: Read> Decoder<R> {
         self.exif_data.as_deref()
     }
 
+    /// Returns collection of raw app data that couldn't be recognized
+    pub fn unknown_data(&self) -> impl Iterator<Item=&'_ [u8]> + '_ {
```

Comment:
Calling this `unknown_data` feels a bit vague.

Also, I'm noticing that the "unknown" part of the name signals a possible backwards compatibility issue: if later versions of the crate do recognize some other app data, will the user have to switch interfaces?

Reply:
Then maybe store all AppData, including known one. So in the future, when crate recognizes more tags, user will switch from custom implementation of "raw" to crate one?

Reply:
Really makes me consider if data from all `App(n)` variants should be collected?

Just as an overview, there's a whole lot of specific cases that would access these and which we currently do not store: https://exiftool.org/TagNames/JPEG.html

In that case however the `Unknown` variant should collect the `n` so that the caller always knowns which kind of app segment the data is from.

Reply:
What about a simple struct? Should be semantically cleaner than sticking to tuples. And with that, fintelias preference in the other comment makes more sense to me, i.e. the public API can return `&[AppSegment]` for accessing them instead of that impl-iterator type.

```rust
struct AppSegment {
    pub kind: u8,
    pub data: Vec<u8>,
}
```

Reply:
Also, the exif data can just be stored as an index in the `Vec<AppSegment>` rather than needing to keep around an extra copy

Reply:
The difference between exif data `[u8]` and app_segment with exif `[u8]` is header `*b"Exif\x00\x00"`.